lib/ads/interstitial_ad_manager.js

```diff
@@ -1630,6 +1645,59 @@ shaka.ads.InterstitialAdManager = class {
   getInterstitials() {
     return Array.from(this.interstitials_);
   }
+
+
+  /**
+   * @param {!shaka.extern.AdInterstitial} interstitial
+   * @private
+   */
+  checkPreloadImages_(interstitial) {
+    if (this.preloadImages_.has(interstitial)) {
+      return;
+    }
+    const links = [];
+    if (interstitial.background) {
+      const urlRegExp = /url\(('|")?([^'"()]+)('|")\)?/;
+      const match = interstitial.background.match(urlRegExp);
+      if (match) {
+        const url = match[2];
+        const link = /** @type {HTMLLinkElement} */(
+          document.createElement('link'));
+        link.rel = 'preload';
+        link.href = url;
+        link.as = 'image';
+        document.head.appendChild(link);
+        links.push(link);
+      }
+    }
+    if (interstitial.mimeType.startsWith('image/')) {
+      const url = interstitial.uri;
+      const link = /** @type {HTMLLinkElement} */(
+        document.createElement('link'));
+      link.rel = 'preload';
+      link.href = url;
+      link.as = 'image';
+      document.head.appendChild(link);
+      links.push(link);
```

Comment:
Those 2 closures are almost the same, please add some helper method.

Reply:
Done!